xonsh/history.py

```diff
@@ -229,6 +232,180 @@ def i_am_at_the_front(self):
+        try:
+            json_file = lazyjson.LazyJSON(f, reopen=False)
+            file_hist.append(json_file.load()['cmds'])
+        except ValueError:
+            # Invalid json file
+            pass
+    commands = [(c['inp'].replace('\n', ''), c['ts'][0])
+                for commands in file_hist for c in commands if c]
+    commands.sort(key=itemgetter(1))
+    return [(c, t, ind) for ind, (c, t) in enumerate(commands)]
+
+
+def _curr_session_formatter(hist):
+    """
+    Take in History object and return command list tuple with
+    format: (name, start_time, index)
+    """
+    if not hist:
+        hist = builtins.__xonsh_history__
+    if not hist:
+        return None
+    start_times = [start for start, end in hist.tss]
+    names = [name[:-1] if name.endswith('\n') else name
+             for name in hist.inps]
+    commands = enumerate(zip(names, start_times))
+    return [(c, t, ind) for ind, (c, t) in commands]
+
+
+def _zsh_hist_formatter(location=None):
+    if not location:
+        location = os.path.join('~', '.zsh_history')
+    z_hist_formatted = list()
+    z_path = os.path.expanduser(location)
+    if os.path.isfile(z_path):
+        with open(z_path, 'r') as z_file:
+            z_txt = z_file.read()
+            z_txt = z_txt.encode('utf-8', 'replace').decode('utf-8')
+            z_hist = z_txt.splitlines()
+            try:
+                if z_hist:
+                    for ind, line in enumerate(z_hist):
+                        start_time, command = line.split(';')
+                        try:
+                            start_time = float(start_time.split(':')[1])
+                        except ValueError:
+                            start_time = -1
+                        z_hist_formatted.append((command, start_time, ind))
+                    return z_hist_formatted
+            except Exception as e:
+                print("There was a problem parsing {}.".format(z_path))
+                raise e
+                return None
+
+    else:
+        print("No zsh history file found at: {}".format(z_path))
+        return None
+
+
+def _bash_hist_formatter(location=None):
+    if not location:
+        location = os.path.join('~', '.bash_history')
+    bash_hist_formatted = list()
+    b_path = os.path.expanduser(location)
+    if os.path.isfile(b_path):
+        try:
+            with open(b_path, 'r') as bash_file:
+                b_txt = bash_file.read()
+                b_txt = b_txt.encode('utf-8', 'replace').decode('utf-8')
+                bash_hist = b_txt.splitlines()
+                if bash_hist:
+                    for ind, command in enumerate(bash_hist):
+                        bash_hist_formatted.append((command, 0.0, ind))
+                    return bash_hist_formatted
+        except:
+            print("There was a problem parsing {}.".format(b_path))
+            return None
+    else:
+        print("No bash history file found at: {}".format(b_path))
+        return None
```

Comment:
I think you can remove the `return None` lines and the flow will work itself out